x/mint/keeper/keeper_test.go

```diff
@@ -719,3 +719,348 @@ func (suite *KeeperTestSuite) TestDistributeDeveloperRewards() {
+			amountDistributed: sdk.NewInt(100),
+
+			expectedDistributedAmount: sdk.OneInt(),
+			expectedPersistedInStore:  sdk.ZeroDec(),
+		},
+		"valid case: 100.8 - 100 + 0.4 = 1.2; 1 distributed, 0.2 persisted in store; mint module account": {
+			moduleAccountName: types.ModuleName,
+			preExistingDelta:  sdk.NewDecWithPrec(4, 1),
+			provisions:        sdk.NewDecCoinFromDec(sdk.DefaultBondDenom, sdk.NewDecWithPrec(1008, 1)),
+			amountDistributed: sdk.NewInt(100),
+
+			expectedDistributedAmount: sdk.OneInt(),
+			expectedPersistedInStore:  sdk.NewDecWithPrec(2, 1),
+		},
+		"valid case: 100 - 100 + 0 = 0; none distributed or persisted in store; mint module account": {
+			moduleAccountName: types.ModuleName,
+			preExistingDelta:  sdk.ZeroDec(),
+			provisions:        sdk.NewDecCoinFromDec(sdk.DefaultBondDenom, sdk.NewDec(100)),
+			amountDistributed: sdk.NewInt(100),
+
+			expectedDistributedAmount: sdk.ZeroInt(),
+			expectedPersistedInStore:  sdk.ZeroDec(),
+		},
+		"invalid module account name - error": {
+			moduleAccountName: poolincentivestypes.ModuleName,
+			preExistingDelta:  sdk.NewDecWithPrec(2, 1),
+			provisions:        sdk.NewDecCoinFromDec(sdk.DefaultBondDenom, sdk.NewDecWithPrec(1006, 1)),
+			amountDistributed: sdk.NewInt(100),
+
+			expectErr: true,
+		},
+		"provisions are less than amount distributed - error": {
+			moduleAccountName: types.ModuleName,
+			preExistingDelta:  sdk.NewDecWithPrec(2, 1),
+			provisions:        sdk.NewDecCoinFromDec(sdk.DefaultBondDenom, sdk.NewDecWithPrec(1006, 1)),
+			amountDistributed: sdk.NewInt(101),
+
+			expectErr: true,
+		},
+		"attempted to distribute more than developer vesting module account balance - error": {
+			moduleAccountName: types.DeveloperVestingModuleAcctName,
+			preExistingDelta:  sdk.NewDecFromInt(sdk.NewInt(keeper.DeveloperVestingAmount)).Add(sdk.OneDec()),
+			provisions:        sdk.NewDecCoinFromDec(sdk.DefaultBondDenom, sdk.NewDecWithPrec(1006, 1)),
+			amountDistributed: sdk.NewInt(100),
+
+			expectErr: true,
+		},
+	}
+	for name, tc := range tests {
+		suite.Run(name, func() {
+			// Setup.
+			suite.Setup()
+			mintKeeper := suite.App.MintKeeper
+			ctx := suite.Ctx
+
+			if !tc.preExistingDelta.IsNil() {
+				mintKeeper.SetTruncationDelta(ctx, tc.moduleAccountName, tc.preExistingDelta)
+			}
+
+			// System under test
+			actualAmount, err := mintKeeper.HandleTruncationDelta(ctx, tc.moduleAccountName, tc.provisions, tc.amountDistributed)
+
+			// Assertions.
+			if tc.expectErr {
+				suite.Require().Error(err)
+				return
+			}
+
+			suite.Require().NoError(err)
+			suite.Require().Equal(tc.expectedDistributedAmount.String(), actualAmount.String())
+			actualPersistedInstore, err := mintKeeper.GetTruncationDelta(ctx, tc.moduleAccountName)
+			suite.Require().NoError(err)
+			suite.Require().Equal(tc.expectedPersistedInStore, actualPersistedInstore)
+		})
+	}
+}
+
+func (suite *KeeperTestSuite) TestSetTruncationDelta() {
```

Comment:
Am I right in saying the truncation delta persisted in store should never be greater than 1? If so, can/should we guarantee that through code/tests?

Reply:
Will resolve this comment because we were discussing this earlier here:
https://github.com/osmosis-labs/osmosis/pull/2503#discussion_r954465923